### PR TITLE
Add alignment option to GridColumn 

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | `GridColumn` add  alignment `option`.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Required here https://github.com/PrestaShop/PrestaShop/pull/23515
| How to test?      | See description below 
| Possible impacts? | N/A

- edit `src/Core/Grid/Definition/Factory/CategoryGridDefinitionFactory.php` in `getColumns` method add `alignment` option in `description` column : 

```php
->add(
    (new DataColumn('description'))
      ->setName($this->trans('Description', [], 'Admin.Global'))
      ->setOptions([
             'field' => 'description',
            'sortable' => false,
            'alignment' => 'center' // new alignment option
      ])
)
```

## Before

image

## After 

- #### Alignment `center`

image

- #### Alignment `right`

image

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24382)

### CODE REVIEW

File: src/Core/Grid/Column/AbstractColumn.php [hunk 1]
```diff
@@ -128,9 +128,12 @@ protected function configureOptions(OptionsResolver $resolver)
             ->setDefaults([
                 'sortable' => true,
                 'clickable' => false,
+                'alignment' => false,
```
Comment (Progi1984) [inline src/Core/Grid/Column/AbstractColumn.php]:
```suggestion
                'alignment' => 'left',
```

File: src/Core/Grid/Column/AbstractColumn.php [hunk 2]
```diff
@@ -128,9 +128,12 @@ protected function configureOptions(OptionsResolver $resolver)
...
             ])
             ->setAllowedTypes('sortable', 'bool')
-            ->setAllowedTypes('clickable', 'bool');
+            ->setAllowedTypes('clickable', 'bool')
+            ->setAllowedTypes('alignment', ['bool', 'string'])
```
Comment (Progi1984) [inline src/Core/Grid/Column/AbstractColumn.php]:
```suggestion
            ->setAllowedTypes('alignment', 'string')
```

File: src/Core/Grid/Column/AbstractColumn.php [hunk 3]
```diff
@@ -128,9 +128,12 @@ protected function configureOptions(OptionsResolver $resolver)
...
+            ->setAllowedValues('alignment', ['center', 'left', 'right', 'justify', false]);
```
Comment (Progi1984) [inline src/Core/Grid/Column/AbstractColumn.php]:
```suggestion
            ->setAllowedValues('alignment', ['center', 'left', 'right', 'justify']);
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 4]
```diff
@@ -51,7 +51,9 @@
         <tr>
           {% for column in grid.columns %}
             <td class="{{ column.type }}-type column-{{ column.id }}{% if column.options.clickable is defined and column.options.clickable %} clickable{% endif %}">
-              {{ column_content(record, column, grid) }}
+              <div {% if column.options.alignment is defined and column.options.alignment %} class="text-{{ column.options.alignment }}" {% endif %}>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]:
```suggestion
              <div class="text-{{ column.options.alignment }}">
```

**Review by Progi1984 — COMMENTED** [review]
As the default alignment is in `left`, please use it.

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig (hunk 4 above)
Comment (PululuK) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]: @Progi1984  some types redefine options, so always check if the option is defined. See for example https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Grid/Column/Type/Category/CategoryPositionColumn.php#L48

**Comment by Progi1984** [comment]
@PululuK I think you should rebase the PR.

**Comment by PululuK** [comment]
> @PululuK I think you should rebase the PR.

Why ?

**Comment by Progi1984** [comment]
@PululuK Because CI is broken and it has been fixed on `1.7.8.x` branch.

**Comment by PululuK** [comment]
> @PululuK Because CI is broken and it has been fixed on `1.7.8.x` branch.

Yes, but in this case We wait for merge, right ?
(I think it is about https://github.com/PrestaShop/PrestaShop/pull/24377)

**Comment by Progi1984** [comment]
@PululuK No, because we can't send your PR in QA without :green_circle: CI.

**Comment by PululuK** [comment]
I'm talking about 1.7.8.x --> develop merge

**Comment by Progi1984** [comment]
Oh. Misseen :stuck_out_tongue:

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 5]
```diff
@@ -51,7 +51,9 @@
         <tr>
           {% for column in grid.columns %}
             <td class="{{ column.type }}-type column-{{ column.id }}{% if column.options.clickable is defined and column.options.clickable %} clickable{% endif %}">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]:
```suggestion
            <td class="{{ column.type }}-type column-{{ column.id }}{% if column.options.clickable is defined and column.options.clickable %} clickable{% endif %}{% if column.options.alignment is defined and column.options.alignment %} text-{{ column.options.alignment }}{% endif %}">
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig (hunk 4 above)
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]:
```suggestion
```

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 6]
```diff
@@ -51,7 +51,9 @@
...
-              {{ column_content(record, column, grid) }}
+              <div {% if column.options.alignment is defined and column.options.alignment %} class="text-{{ column.options.alignment }}" {% endif %}>
+                  {{ column_content(record, column, grid) }}
+              </div>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]:
```suggestion
```

**Review by Progi1984 — COMMENTED** [review]
A last improvement for avoiding an unuseful `div`

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 7]
```diff
@@ -50,8 +50,8 @@
       {% for record in grid.data.records %}
         <tr>
           {% for column in grid.columns %}
-            <td class="{{ column.type }}-type column-{{ column.id }}{% if column.options.clickable is defined and column.options.clickable %} clickable{% endif %}">
-              {{ column_content(record, column, grid) }}
+            <td class="{{ column.type }}-type column-{{ column.id }}{% if column.options.clickable is defined and column.options.clickable %} clickable{% endif %}{% if column.options.alignment is defined and column.options.alignment %} text-{{ column.options.alignment }}{% endif %}">
+                  {{ column_content(record, column, grid) }}
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig]:
```suggestion
              {{ column_content(record, column, grid) }}
```

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]
Looks fine to me, cool :)

**Review by kpodemski — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM !! Thanks @PululuK !!

**Comment by Progi1984** [comment]
Thanks @PululuK & @SD1982

**Pull request merged**